Flag Duplicate & Mutually Exclusive Compiler Opts

montymxb opened the pull request:
Closes #105 by adding:
- warning diagnostic for duplicate options `(AG, AG)`
- warning diagnostic for mutually exclusive options `(NOEXIT, EXIT)`
- warning for dupes in short & long form `(AG, AGGREGATE)`
- warning for mutually exclusive options in long & short form `(NAG, AGGREGATE)`
- updated the existing diagnostic for unrecognized options to conform to PLI's compiler code for that
- updated `assertNoDiagnostics` to factor in compiler option diagnostics as well

In all cases the compiler only emits a warning rather than an error. And in cases of duplicate / mutex options the compiler always appears to take the last option seen (verified in listings output).

ssmifi commented on packages/language/src/preprocessor/compiler-options/translator.ts:
```diff
@@ -103,9 +108,36 @@ class Translator {
+   */
+  checkDuplicateOption(option: CompilerOption, name: string): void {
+    const modifiedName = name.startsWith("NO") ? name.slice(2) : `NO${name}`;
+    if (this.optionSet.has(name)) {
```
_Question:_ Works for the cases you've listed in the PR. However, I was wondering, if we also want to support long vs short form and short form negation? E.g., AGGREGATE, AG and AG, NAG

montymxb replied:
Good point. I haven't double checked those short hand aliases yet, or there non-standard forms. I take care of that in a second pass, and get some tests in for those edge cases as well.

ssmifi reviewed (commented):
LGTM. One question if we should support the short forms of compiler options and their negations.

montymxb commented:
Refactored to rely on the pre-existing TranslatorRules for tracking which options have been applied. Updated tests to check for short & long hand forms in turn, and updated the `assertNoDiagnostics` function to fail on compiler option diagnostics as well.

msujew reviewed (approved):
Great, looks good to me 👍
